Magento2 resizes and recompresses original image during upload #17910

mahsn opened the pull request:
**Preconditions**
2.2.5 and below versions are affected
**Steps to reproduce**
Create jpg file larger than 1900x1200 pixels

1. Login to admin
2. Create or update existing product image by uploading new file you created
3. Observe that after upload image is resized to not exceed 1900x1200
4. File is also processed by php using compression quality setting
5. Expected result
6. File is uploaded as it is - it is not modified by Magento 2

**Actual result**
File is resized
File is recompressed
This issue dublicate  with #13747 , there is some features that already added in magento 2.2. I created 3 input in Configuration > Advanced > Developer under the Image Processing Settings to put max values of images.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

Commit:
Merge branch '2.3-develop' of https://github.com/magento/magento2ce into MAGETWO-93969

# Conflicts:
#	app/etc/di.xml

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

Commit:
merge magento/2.3-develop into magento-honey-badgers/MAGETWO-91439-Price-prices-disappearing-on-category-page-remove-resolver

Commit:
Merge branch '2.3-develop' of https://github.com/magento/magento2ce into MAGETWO-93969

# Conflicts:
#	app/code/Magento/Checkout/Controller/Cart/UpdatePost.php
#	dev/tests/integration/testsuite/Magento/Customer/Controller/Adminhtml/Index/MassAssignGroupTest.php
#	dev/tests/integration/testsuite/Magento/Customer/Controller/Adminhtml/Index/MassDeleteTest.php

rogyar committed:
Api-functional tests added

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

Commit:
Merge branch '2.3-develop' of https://github.com/magento/magento2ce into MAGETWO-93969

cpartica committed:
MAGETWO-91439: Price prices disappearing on category page

- keep order or generated vs cached

Commit:
merge magento/2.3-develop into magento-honey-badgers/MAGETWO-91439-Price-prices-disappearing-on-category-page-remove-resolver

rogyar committed:
Check that rendered values do not contain directives

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

Commit:
Merge branch '2.3-develop' of https://github.com/magento/magento2ce into MAGETWO-93969

# Conflicts:
#	app/code/Magento/CheckoutAgreements/Controller/Adminhtml/Agreement/Delete.php
#	app/code/Magento/CheckoutAgreements/Controller/Adminhtml/Agreement/Edit.php
#	app/code/Magento/CheckoutAgreements/Controller/Adminhtml/Agreement/Save.php

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

Commit:
Merge branch '2.3-develop' of https://github.com/magento/magento2ce into MAGETWO-93969

# Conflicts:
#	dev/tests/integration/testsuite/Magento/Catalog/Controller/Product/CompareTest.php
#	dev/tests/integration/testsuite/Magento/Contact/Controller/IndexTest.php

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

Commit:
Merge branch '2.3-develop' of https://github.com/magento/magento2ce into MAGETWO-93969

# Conflicts:
#	dev/tests/integration/testsuite/Magento/Customer/Controller/AccountTest.php

austris-argalis committed:
Add change customer password mutation

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

Commit:
Merge branch '2.3-develop' of https://github.com/magento/magento2ce into MAGETWO-93969

# Conflicts:
#	app/code/Magento/Sales/Controller/Adminhtml/Order/Status/Save.php
#	app/code/Magento/UrlRewrite/Block/Edit.php

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

Commit:
MAGETWO-91439: Prices disappearing when product is assigned to a different store and default store is disabled
- Fix Functional test

Commit:
MAGETWO-91439: Prices disappearing when product is assigned to a different store and default store is disabled
- Removed commeneted lines

austris-argalis committed:
Rename global mutation, add test case for incorrect 'current' password

Commit:
Merge branch '2.3-develop' of https://github.com/magento/magento2ce into MAGETWO-93969

# Conflicts:
#	dev/tests/integration/testsuite/Magento/Customer/Controller/AddressTest.php
#	dev/tests/integration/testsuite/Magento/Customer/Controller/Adminhtml/IndexTest.php

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

Commit:
MAGETWO-91439: Prices disappearing when product is assigned to a different store and default store is disabled
- Updated schema location for functional test

Commit:
Merge branch '2.3-develop' into MAGETWO-93969

Commit:
MAGETWO-91439: Prices disappearing when product is assigned to a different store and default store is disabled
- Removed schema location update as this test is already in mainline and should be updated part of Pangolins fix

Commit:
Merge branch '2.3-develop' into MAGETWO-93969

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

Commit:
Merge branch '2.3-develop' of https://github.com/magento/magento2ce into MAGETWO-93969

Commit:
Merge branch 'MAGETWO-93969' of https://github.com/magento-qwerty/magento2ce into MAGETWO-93969

Commit:
Merge branch '2.3-develop' of https://github.com/magento/magento2ce into MAGETWO-93969

Commit:
Merge branch '2.3-develop' of https://github.com/magento/magento2ce into MAGETWO-93969

cpartica committed:
Merge remote-tracking branch 'mainline/2.3-develop' into MAGETWO-91439-Price-prices-disappearing-on-category-page-remove-resolver

# Conflicts:
#	app/code/Magento/CatalogUrlRewrite/Plugin/Store/Block/Switcher.php
#	app/code/Magento/Store/Block/Switcher.php
#	app/code/Magento/Store/Controller/Store/SwitchAction.php
#	app/code/Magento/Store/Model/Plugin/StoreCookie.php
#	app/code/Magento/Store/Test/Unit/Model/Plugin/StoreCookieTest.php

rogyar committed:
Cart coupons prototype

rogyar committed:
Merge branch 'mutations-empty-cart' into mutations-coupons

Commit:
Merge branch '2.3-develop' of https://github.com/magento/magento2ce into MAGETWO-93969

# Conflicts:
#	app/code/Magento/Review/Controller/Product/ListAjax.php

chrom committed:
GraphQl-129: Retrieve Customer token

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

Commit:
Merge branch '2.3-develop' of https://github.com/magento/magento2ce into MAGETWO-93969

# Conflicts:
#	dev/tests/integration/testsuite/Magento/Checkout/Controller/CartTest.php

shiftedreality committed:
MAGETWO-94349: The "recursion detected" error during a deployment

shiftedreality committed:
MAGETWO-94349: The "recursion detected" error during a deployment

shiftedreality committed:
Merge branch '2.3-develop' of https://github.com/magento/magento2ce into MAGETWO-94349

Commit:
Merge branch '2.3-develop' of https://github.com/magento/magento2ce into MAGETWO-93969

# Conflicts:
#	lib/internal/Magento/Framework/App/FrontController.php

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

shiftedreality committed:
Merge branch '2.3-develop' of https://github.com/magento/magento2ce into MAGETWO-94349

svitja committed:
MAGETWO-94329: Unable to view order Placed via Checked Out with Multiple Addressees

rogyar committed:
Introduced service for checking cart mutations permissions

rogyar committed:
Use corresponding service for converting masked id to int id

rogyar committed:
Fixed typo for exception message

Commit:
merge magento/2.3-develop into magento-thunder/MAGETWO-94349

svitja committed:
Merge remote-tracking branch 'origin/2.3-develop' into MAGETWO-94329

svitja committed:
Merge remote-tracking branch 'origin/2.3-develop' into MAGETWO-94329

zakdma committed:
MAGETWO-93195: [MFTF] AdminProductBundleCreationTest fails

Commit:
Merge branch '2.3-develop' into MAGETWO-93969

zakdma committed:
Merge remote-tracking branch 'origin/2.3-develop' into MAGETWO-93195

rogyar committed:
Quote masked id creation logic moved to create cart resolver

cpartica committed:
Merge remote-tracking branch 'origin/2.3-develop' into MAGETWO-91439-Price-prices-disappearing-on-category-page-remove-resolver

cpartica committed:
MAGETWO-91439: Prices disappearing when product is assigned to a different store and default store is disabled

- remove file that should of have been removed by merge

svitja committed:
Merge remote-tracking branch 'origin/2.3-develop' into MAGETWO-94329

svitja committed:
Merge remote-tracking branch 'origin/2.3-develop' into MAGETWO-94329

zakdma committed:
Merge remote-tracking branch 'origin/2.3-develop' into MAGETWO-93195

svitja committed:
Merge remote-tracking branch 'origin/2.3-develop' into MAGETWO-94329

OlgaVasyltsun committed:
MAGETWO-58144: [FT] Magento\Catalog\Test\TestCase\Product\ProductTypeSwitchingOnCreationTest fail on Bamboo

zakdma committed:
Merge remote-tracking branch 'origin/2.3-develop' into MAGETWO-93195

zakdma committed:
MAGETWO-93031: Update Magento logo in emails and any other places

zakdma committed:
MAGETWO-93031: Update Magento logo in emails and any other places

cpartica committed:
Merge remote-tracking branch 'mainline/2.3-develop' into MAGETWO-91439-Price-prices-disappearing-on-category-page-remove-resolver

pfantini committed:
GraphQL-129: Change generate token schema and add graphql exception

cpartica committed:
Merge remote-tracking branch 'mainline/2.3-develop' into MAGETWO-91439-Price-prices-disappearing-on-category-page-remove-resolver

svitja committed:
Merge remote-tracking branch 'origin/2.3-develop' into MAGETWO-58144

DianaRusin committed:
Merge remote-tracking branch 'origin/2.3-develop' into MAGETWO-94329

DianaRusin committed:
Merge remote-tracking branch 'origin/MAGETWO-94329' into 2.3-develop-pr34

zakdma committed:
Merge remote-tracking branch 'origin/2.3-develop' into MAGETWO-93195

zakdma committed:
Merge remote-tracking branch 'origin/2.3-develop' into MAGETWO-93031

dthampy committed:
MAGETWO-91439: Prices disappearing when product is assigned to a different store and default store is disabled

- added wait step

zakdma committed:
Merge remote-tracking branch 'origin/2.3-develop' into MAGETWO-93031

zakdma committed:
Merge remote-tracking branch 'origin/2.3-develop' into MAGETWO-93195

Commit:
Merge remote-tracking branch 'origin/2.3-develop' into hashed-cart-create-refactor

Commit:
Merge remote-tracking branch 'origin/2.3-develop' into mutations-coupons

krissyhiserote committed:
MAGETWO-94402: [2.3.0] PayPal Billing Address for Registered Customers

- make sure the require billing address setting is set before getting billing address from PayPal

Commit:
merge magento/2.3-develop into magento-honey-badgers/MAGETWO-91439-Price-prices-disappearing-on-category-page-remove-resolver

Commit:
merge magento/2.3-develop into magento-honey-badgers/MAGETWO-91439-Price-prices-disappearing-on-category-page-remove-resolver

mage2pratik committed:
Replace floatval() function by using direct type casting to (float)

Commit:
merge magento/2.3-develop into magento-honey-badgers/MAGETWO-91439-Price-prices-disappearing-on-category-page-remove-resolver

pfantini committed:
GraphQL-129. Add generate customer token test

OlgaVasyltsun committed:
Merge remote-tracking branch 'origin/2.3-develop' into MAGETWO-93195

OlgaVasyltsun committed:
Merge remote-tracking branch 'origin/MAGETWO-93195' into 2.3-develop-pr34

Commit:
merge magento/2.3-develop into magento-honey-badgers/MAGETWO-91439-Price-prices-disappearing-on-category-page-remove-resolver

pfantini committed:
GrapgQL-129: Add composer dependency and fix codestyle issues

krissyhiserote committed:
MAGETWO-94402: [2.3.0] PayPal Billing Address for Registered Customers

- fix express checkout for logged in customer and code formatting

krissyhiserote committed:
MAGETWO-94402: [2.3.0] PayPal Billing Address for Registered Customers

- fix code formatting

krissyhiserote committed:
Merge remote-tracking branch 'mainline/2.3-develop' into MAGETWO-94402-paypal-billing-address

TBlindaruk committed:
[2.3] Changed intval($val) to (int) $val, since it is faster:
 - changed intval($val) to (int) $val, since it is faster

krissyhiserote committed:
MAGETWO-94402: [2.3.0] PayPal Billing Address for Registered Customers

- add and fix integration tests for PayPal Express Checkout

krissyhiserote committed:
MAGETWO-94402: [2.3.0] PayPal Billing Address for Registered Customers

- fix character length

Commit:
merge magento/2.3-develop into magento-honey-badgers/MAGETWO-91439-Price-prices-disappearing-on-category-page-remove-resolver

cpartica committed:
MAGETWO-91439: Prices disappearing when product is assigned to a different store and default store is disabled

- fix static

zakdma committed:
Merge remote-tracking branch 'origin/2.3-develop' into 2.3-develop-pr34

zakdma committed:
Merge remote-tracking branch 'origin/MAGETWO-93031' into 2.3-develop-pr34

yaroslav-zenin committed:
typofix: ImportCollection -> ItemCollection

Commit:
Merge branch '2.3-develop' of https://github.com/magento/magento2ce into MAGETWO-93969

zakdma committed:
Merge remote-tracking branch 'origin/2.3-develop' into MAGETWO-94329

zakdma committed:
MAGETWO-94329: Unable to view order Placed via Checked Out with Multiple Addressees

zakdma committed:
MAGETWO-58144: [FT] Magento\Catalog\Test\TestCase\Product\ProductTypeSwitchingOnCreationTest fail on Bamboo

zakdma committed:
Merge remote-tracking branch 'origin/2.3-develop' into MAGETWO-58144

zakdma committed:
MAGETWO-58144: [FT] Magento\Catalog\Test\TestCase\Product\ProductTypeSwitchingOnCreationTest fail on Bamboo

magento-engcom-team committed:
ENGCOM-2968: [2.3] Changed intval($val) to (int) $val, since it is faster: #18035

 - Merge Pull Request magento/magento2#18035 from TBlindaruk/magento2:changed_intval_to_int_v2
 - Merged commits:
   1. 90c5387a15ef51c146cc0bc40585ee27af9b1445

zakdma committed:
Merge remote-tracking branch 'origin/MAGETWO-94329' into 2.3-develop-pr34

magento-engcom-team committed:
ENGCOM-2970: Replace floatval() function by using direct type casting to (float) #18020

 - Merge Pull Request magento/magento2#18020 from mage2pratik/magento2:forwardport-patch-16848
 - Merged commits:
   1. fd0e8b383ce7dd586465d450f74d14dfcde4b52f

zakdma committed:
Merge remote-tracking branch 'origin/MAGETWO-58144' into 2.3-develop-pr34

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

Commit:
Merge branch '2.3-develop' of https://github.com/magento/magento2ce into MAGETWO-93969

zakdma committed:
Merge remote-tracking branch 'origin/2.3-develop' into 2.3-develop-pr34

rogyar committed:
Added integration tests for gift message quote merge

cpartica committed:
MAGETWO-91439: Prices disappearing when product is assigned to a different store and default store is disabled

- fix static

Commit:
merge magento/2.3-develop into magento-honey-badgers/MAGETWO-91439-Price-prices-disappearing-on-category-page-remove-resolver

krissyhiserote committed:
MAGETWO-94402: [2.3.0] PayPal Billing Address for Registered Customers

- remove unused variable from tests and avoid using type coercion

krissyhiserote committed:
Merge remote-tracking branch 'mainline/2.3-develop' into MAGETWO-94402-paypal-billing-address

krissyhiserote committed:
MAGETWO-94402: [2.3.0] PayPal Billing Address for Registered Customers

- fix docblocks

mage2pratik committed:
Code improvement of lib files

Commit:
Merge remote-tracking branch 'origin/MAGETWO-94172-ElasticSearch-Index-Exception' into BugFixPR

cpartica committed:
Merge remote-tracking branch 'mainline/2.3-develop' into MAGETWO-91439-Price-prices-disappearing-on-category-page-remove-resolver

Commit:
Merge remote-tracking branch 'origin/MAGETWO-94172-ElasticSearch-Index-Exception' into BugFixPR

pfantini committed:
GraphQL-129: Change resolve to return array, update schema and test

magento-engcom-team committed:
ENGCOM-2972: Code improvement of lib files #18049

 - Merge Pull Request magento/magento2#18049 from mage2pratik/magento2:code-improvement-7
 - Merged commits:
   1. 5ef0489ac2504bd089e3f176fa5995a1eb90ef92

Commit:
ENGCOM-2323: [Forwardport] #7903 correct the position of the datepicker when you scroll #16776

Commit:
ENGCOM-1534: Add a link to the cart to the success message when adding a product (Magento 2.3) #14059

Commit:
ENGCOM-1391: [2.3-develop] Forwardport of magento/magento2#11323 #184

ishakhsuvarov committed:
magento/magento2#15947: Merge branch '2.3-develop' of github.com:magento/magento2 into 2.3-develop-PR-port-12935

ishakhsuvarov committed:
Add Ability To Separate Frontend / Adminhtml in New Relic
 - Minor improvements

Commit:
Merge branch '2.3-develop' of https://github.com/magento/magento2ce into MAGETWO-93969

yaroslav-zenin committed:
Merge branch '2.3-develop' of github.com:magento/magento2 into 2.3-develop

Commit:
magento-engcom/magento2ce#2176: fixed failed tests

ishakhsuvarov committed:
Add Ability To Separate Frontend / Adminhtml in New Relic
 - Minor improvements

magento-engcom-team committed:
ENGCOM-2974: [Forwardport] Add Ability To Separate Frontend / Adminhtml in New Relic #15947

 - Merge Pull Request magento/magento2#15947 from ihor-sviziev/magento2:2.3-develop-PR-port-12935
 - Merged commits:
   1. 92d8f138e9f1a7d2b05bb18ffb4e4343bb2fe99d
   2. e8965a82388dedd94c77ead0bd2789c7e39d25bd
   3. 0d169ac0952d77459c011ccbb08ca1c5cc9800c9
   4. 4d8c70224079604ce56b484c70f90552a85df450
   5. 630ea11e0f2a2ea1e838cd827363d4696022d533
   6. 9fa2765bb96f5fe737d809bf4d6b993a381135fa
   7. 5c607a4f83bd5799f95b886274e2493d012d7d96
   8. 009082b3f229a92c95a883c7f06dcc3d1fe9ad6b
   9. 061a5a102c100af60e59101b6b33ff1d679f542c
   10. 05e2d82ec55cb97448c246f793dcf6461670b273
   11. 7fde1e786b0a69c73a99c8468f1a538a9006cf54
   12. b8e255864f226eea060e3afd628f8ceb5d58a1df
   13. ccc355a63f5c17222466834113fbde599af19290
   14. b9e20c7fc508d0272ca1916a8e89cfd74a038e94

TomashKhamlai committed:
Remove PHPMD directive

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

zakdma committed:
Merge remote-tracking branch 'origin/2.3-develop' into 2.3-develop-pr34

TomashKhamlai committed:
Merge remote-tracking branch 'origin/2.3-develop' into recheck-129-retrive-customer-token

TomashKhamlai committed:
Simplify code to avoid complexity of getting just one string

TomashKhamlai committed:
Remove usage of Value Factory and optimize imports

magento-engcom-team committed:
ENGCOM-2979: Added integration test for gift message quote merge #18047

 - Merge Pull Request magento/magento2#18047 from rogyar/magento2:2.3-quote-merge-giftmessage-it
 - Merged commits:
   1. e2d96ae6a374eec4cba5a4418f2274067633c791

TomashKhamlai committed:
Add checks for input arguments

Commit:
Merge remote-tracking branch 'upstream/2.3-develop' into MAGETWO-94402-paypal-billing-address

okolesnyk committed:
ENGCOM-1534: Add a link to the cart to the success message when adding a product (Magento 2.3) #14059

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.3-develop

 - merged latest code from mainline branch

Commit:
Merge remote-tracking branch 'upstream/2.3-develop' into BugFixPR

Commit:
Merge remote-tracking branch 'owls/MAGETWO-94402-paypal-billing-address' into BugFixPR

okolesnyk committed:
ENGCOM-1534: Add a link to the cart to the success message when adding a product (Magento 2.3) #14059

okolesnyk committed:
ENGCOM-1534: Add a link to the cart to the success message when adding a product (Magento 2.3) #14059

Commit:
Merge remote-tracking branch 'upstream/2.3-develop' into BugFixPR

ishakhsuvarov committed:
Merge branch 'ENGCOM-2974-magento-magento2-15947' of github.com:magento-engcom/magento2ce into 2.3-develop-prs

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.3-develop

 - merged latest code from mainline branch

cpartica committed:
Merge remote-tracking branch 'mainline/2.3-develop' into MAGETWO-91439-Price-prices-disappearing-on-category-page-remove-resolver

Commit:
ENGCOM-2323: [Forwardport] #7903 correct the position of the datepicker when you scroll #16776

pfantini committed:
GraphQL-80: rename canonical_url to relative_url and fix covered test

pfantini committed:
GraphQL-80: remove extra lines

dmytro-ch committed:
Sales: add missing unit tests for model classes

Commit:
GraphQl-44: added schema for format of textarea ui element

Signed-off-by: vitaliyboyko <v.boyko@atwix.com>

Commit:
Merge branches '2.3-develop' and 'rendered-attributes-product' of https://github.com/magento/graphql-ce into rendered-attributes-product

Commit:
merge magento/2.3-develop into magento-tsg/2.3-develop-pr34

magento-engcom-team committed:
ENGCOM-2984: Sales: add missing unit tests for model classes #18072

 - Merge Pull Request magento/magento2#18072 from dmytro-ch/magento2:2.3-unit-test-sales-model
 - Merged commits:
   1. ec7e74cbcd1f30f94a0aeed5d468a26cc1994081

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.3-develop

 - merged latest code from mainline branch

Commit:
merge magento/2.3-develop into magento-qwerty/MAGETWO-93969

Commit:
[Revert]GraphQl-44: added schema for format of textarea ui element

Will be added in scope of further PR's

Signed-off-by: vitaliyboyko <v.boyko@atwix.com>

Commit:
Magento2 resizes and recompresses original image during upload #17910

Commit:
Merge branch '2.3-develop' of github.com:mahsn/magento2 into 2.3-develop

Commit:
Removing sample data classes

Commit:
Removing sample data

Commit:
Removing sampledata

magento-engcom-team commented:
Hi @mahsn. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me $VERSION instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento/magento2?pullRequest=18087) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=18087) before we can accept your contribution.<br/>

Commit:
Merge pull request #3173 from magento-engcom/2.3-develop-prs

[EngCom] Public Pull Requests - 2.3-develop

Commit:
Merge remote-tracking branch 'upstream/2.3-develop' into BugFixPR

cpartica committed:
Merge pull request #3170 from magento-borg/BugFixPR

[2.3.0-regression] Bug fixes

Commit:
merge magento/2.3-develop into magento-honey-badgers/MAGETWO-91439-Price-prices-disappearing-on-category-page-remove-resolver

cpartica committed:
Merge pull request #3115 from magento-honey-badgers/MAGETWO-91439-Price-prices-disappearing-on-category-page-remove-resolver

[honey] MAGETWO-91439: Prices disappearing when product is assigned to a different store and default store is disabled

zakdma committed:
Merge remote-tracking branch 'origin/2.3-develop' into 2.3-develop-pr34

Commit:
Merge branch '2.3-develop' of https://github.com/magento/magento2ce into MAGETWO-93969

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

magento-engcom-team committed:
ENGCOM-2969: typofix: ImportCollection -> ItemCollection #18043

 - Merge Pull Request magento/magento2#18043 from yaroslav-zenin/magento2:2.3-develop
 - Merged commits:
   1. 608d825eba3c976b19cfe49d77e4ac9fbc4c63ae
   2. 2051521670984913686d21d702b21cb1cfa9a067

Commit:
ENGCOM-2984: Sales: add missing unit tests for model classes #18072

Commit:
ENGCOM-2979: Added integration test for gift message quote merge #18047

Commit:
ENGCOM-2972: Code improvement of lib files #18049

Commit:
ENGCOM-2970: Replace floatval() function by using direct type casting to (float) #18020

Commit:
ENGCOM-2968: [2.3] Changed intval($val) to (int) $val, since it is faster: #18035

Commit:
ENGCOM-2969: typofix: ImportCollection -> ItemCollection #18043

Commit:
GraphQL-140: Added HTML renderer for description and short description

magento-engcom-team committed:
ENGCOM-2991: GraphQL-80: rename canonical_url to relative_url and fix covered test #185

 - Merge Pull Request magento/graphql-ce#185 from pfantini/graphql-ce:80-urlResolver-rename-canonical_url-to-relative_url
 - Merged commits:
   1. c7635146f3d70e8b69230ac8d89493d88367cd4d
   2. 184f90b495ddf0475a34adab7a53250405ba562c

Commit:
Merge remote-tracking branch 'origin/2.3-develop' into recheck-129-retrive-customer-token

magento-engcom-team committed:
ENGCOM-2989: Recheck 129 retrive customer token #184

 - Merge Pull Request magento/graphql-ce#184 from magento/graphql-ce:recheck-129-retrive-customer-token
 - Merged commits:
   1. 527e4febe49ff6c2ab3b42ddabf798307f0e7222
   2. ea9b2a8a73da00df665da807b830615c129945fc
   3. ca0c135c56511a3b67319f54f1ce39894c764e55
   4. c0d1b7b7400dd53e78e15b42d68c6bf4df46b195
   5. e97301d483ed196eef7b7ac33f3a26bc1d737ed9
   6. 85cd23901cd2e4cbdd184a172358314f7094777e
   7. 6a250c182293b6e209b78500ae771ffbeced570f
   8. 21e719c2f7062652b8b1d564474bac1426242633
   9. ad388f6e1ed3ee97c27220066938c9efa9561837
   10. 277fe084c5af746dbdc254d9ffb5929c175a0848
   11. 5dce3bd8eeb8c93b5cb347cd9d2dcb9f2e688bae

Commit:
Merge remote-tracking branch 'origin/2.3-develop' into rendered-attributes-product

TomashKhamlai committed:
Merge remote-tracking branch 'origin/2.3-develop' into issue-54

TomashKhamlai committed:
Add checks for input parameters

rogyar committed:
Merge branch '2.3-develop' into mutations-coupons

rogyar committed:
Implementation improvements

rogyar committed:
Merge branch 'mutations-coupons' of github.com:magento/graphql-ce into mutations-coupons

Commit:
Merge pull request #3164 from magento-tsg/2.3-develop-pr34

[TSG] Upporting for 2.3 (pr34) (2.3.0)

Commit:
MAGETWO-93969: Declaring allowed HTTP methods for controllers

Commit:
merge magento/2.3-develop into magento-qwerty/MAGETWO-93969

phoenix128 commented on lib/internal/Magento/Framework/File/Uploader.php:
```diff
@@ -140,14 +140,14 @@ class Uploader
      * @see \Magento\Framework\Image\Adapter\UploadConfigInterface::getMaxWidth()
      */
-    const MAX_IMAGE_WIDTH = 4096;
+    const MAX_IMAGE_WIDTH = 'dev/image/max_width';
```
You should not replace a constant value in this way. Since it is public a 3rd party extension could have used it (even if it is deprecated now).

mahsn replied:
hi @phoenix128  , thanks for your review , i am going to fix that

Commit:
GraphQL-169: Quote masked id creation logic moved to create cart resolver

magento-engcom-team committed:
ENGCOM-2994:  Quote masked id creation logic moved to create cart resolver #169

 - Merge Pull Request magento/graphql-ce#169 from magento/graphql-ce:hashed-cart-create-refactor
 - Merged commits:
   1. 8771f41e6191d3a7e26cca0413740524f1106b23
   2. 8b6aab66d7e34a2794a5e247545cdc37375d0d14
   3. afa216974bf1214ee860e7bb9eeaf0474aee1d4f

dvoskoboinikov committed:
Merge pull request #3174 from magento-qwerty/MAGETWO-93969

Performed tasks:
- MAGETWO-93969 Declaring allowed HTTP methods for controllers

shiftedreality committed:
Merge branch '2.3-develop' of https://github.com/magento/magento2ce into MAGETWO-94349

shiftedreality committed:
Merge remote-tracking branch 'Thunder/MAGETWO-94349' into MAGETWO-94349

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.3-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
ENGCOM-2990: Added HTML renderer for description and short description #140

 - Merge Pull Request magento/graphql-ce#140 from magento/graphql-ce:rendered-attributes-product
 - Merged commits:
   1. 0cdfdcb5676bd24a3ab2562ddd59501868c887c1
   2. 6638f7eb8a5754ca71e33220d468f365c98c6fa4
   3. 4fbb2aee99f5e1295f00e7a71ab36049ecf0abb2
   4. df0d2ac5cfe4cb3cfc5333b4bfd254333a0a87d5
   5. 4b14faa26ada4bd3e226937044254de3a9fd7b0a
   6. 506c1482629ca2f2a59adc7f99fff272c5a82ad8
   7. 854264d899b5cc1616d0889c8f67b96385716095
   8. 3f92af0cbc88784f77c58028d8a028f87a665fd2

Commit:
magento-engcom/magento2ce#2180: Code style fixes

Commit:
ENGCOM-2994:  Quote masked id creation logic moved to create cart resolver #169

Commit:
ENGCOM-2991: GraphQL-80: rename canonical_url to relative_url and fix covered test #185

Commit:
ENGCOM-2990: Added HTML renderer for description and short description #140

Commit:
ENGCOM-2989: Recheck 129 retrive customer token #184

TomashKhamlai committed:
Import long description

Commit:
GraphQL-64: [Mutations] Cart Operations > Coupons

Commit:
Merge remote-tracking branch 'origin/2.3-develop' into mutations-coupons

Commit:
Merge remote-tracking branch 'origin/2.3-develop' into issue-54

Commit:
GraphQL-64: [Mutations] Cart Operations > Coupons

-- fix static tests

cpartica committed:
Merge pull request #3100 from magento-thunder/MAGETWO-94349

[Lightning] MAGETWO-94349: The "recursion detected" error during a deployment

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.3-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.3-develop

 - merged latest code from mainline branch

ishakhsuvarov committed:
:arrows_clockwise: [EngCom] Public Pull Requests - 2.3-develop

Accepted Public Pull Requests:
 - magento/magento2#18072: Sales: add missing unit tests for model classes (by @dmytro-ch)
 - magento/magento2#18047: Added integration test for gift message quote merge (by @rogyar)
 - magento/magento2#18049: Code improvement of lib files (by @mage2pratik)
 - magento/magento2#18020: Replace floatval() function by using direct type casting to (float) (by @mage2pratik)
 - magento/magento2#18035: [2.3] Changed intval($val) to (int) $val, since it is faster: (by @TBlindaruk)
 - magento/magento2#18043: typofix: ImportCollection -> ItemCollection (by @yaroslav-zenin)

magento-engcom-team committed:
[EngCom] Public Pull Requests - 2.3-develop

 - merged latest code from mainline branch

Commit:
Merge pull request #3183 from magento-engcom/graphql-develop-prs

[EngCom] Public Pull Requests - GraphQL

eduard13 commented on app/code/Magento/Backend/etc/config.xml:
```diff
@@ -47,6 +43,13 @@
+        <dev>
+            <image>
+                <max_quality>4096</max_quality>
+                <max_width>4096</max_width>
```
It's better to keep the previous values as default ones. Later the admin can change them based on his needs.

mahsn replied:
hi @eduard13  , thanks for your review, i thought that using the max values as default values is going to be better,

eduard13 commented on app/code/Magento/Backend/etc/config.xml:
```diff
@@ -47,6 +43,13 @@
         </web>
+        <dev>
+            <image>
+                <max_quality>4096</max_quality>
```
You can't set any value as the max quality. The higher quality that you can use is 100, so 4096 - isn't a very good default value.

eduard13 commented on app/code/Magento/Backend/etc/adminhtml/system.xml:
```diff
@@ -203,6 +203,21 @@
+                <field id="max_quality" translate="label comment" type="text" sortOrder="100" showInDefault="1" showInWebsite="0" showInStore="0" canRestore="1">
+                    <label>Maximum Quality</label>
+                    <validate>validate-greater-than-zero validate-number required-entry</validate>
+                    <comment>Maximum allowed width for uploaded image.</comment>
```
Change the field's comment based on the field's purpose. Also you should mention that this quality is only for products and only for JPEG images.

eduard13 commented on app/code/Magento/Backend/etc/adminhtml/system.xml:
```diff
@@ -203,6 +203,21 @@
                     <backend_model>Magento\Config\Model\Config\Backend\Image\Adapter</backend_model>
                     <comment>When the adapter was changed, please flush Catalog Images Cache.</comment>
                 </field>
+                <field id="max_quality" translate="label comment" type="text" sortOrder="100" showInDefault="1" showInWebsite="0" showInStore="0" canRestore="1">
```
Here you're just adding a new configuration field, I don't see the implementation/usage of this field.

mahsn replied:
Actually i try to use max quality for images, but i couldn't find how to use that in magento 2

eduard13 reviewed (changes requested):
@mahsn I've left several comments. Please review them.
Thanks.

Commit:
Merge remote-tracking branch 'origin/2.3-develop' into issue-54

# Conflicts:
#	app/code/Magento/CustomerGraphQl/etc/schema.graphqls

Commit:
GraphQL-54: [Mutations] My Account: Change Password

Commit:
GraphQL-54: [Mutations] My Account: Change Password

Commit:
GraphQL-54: [Mutations] My Account: Change Password

rogyar committed:
Fixed error messages assertions in api-functional tests

rogyar committed:
Fixed coupon fixture

Commit:
GraphQL-54: [Mutations] My Account: Change Password

Commit:
GraphQL-54: [Mutations] My Account: Change Password

rogyar committed:
Use another fixture for api-functional tests

magento-engcom-team committed:
ENGCOM-3004: [Recheck] Allow changing logged in customers password with Graphql mutation #187

 - Merge Pull Request magento/graphql-ce#187 from magento/graphql-ce:issue-54
 - Merged commits:
   1. e745e9170868749dc8a4c479c73ead9305325fe5
   2. e6043eb7b474478204a7abf75f7dbc51b8e4735e
   3. 8cb23648059190b99570f4982dfc9f8014a3d356
   4. 5d436f10720d15e6f85032e019566d6824c25f29
   5. 23009eb96f9d3025bc92e1164a141e96008d614e
   6. 38a14b8fc56ba68081a4d15f3b661b17f2d09117
   7. a17d437dfdd8958bf842c813d677ade69e8fbf23
   8. 73404e3b37462682fe26e88fe9d42a082381026c
   9. 20e9730fa17adf3fae9421ca58dd3bec25124619
   10. 611d78cd0c044ffa6895d5ac7271fff2ddbf9145
   11. 67145c340543cdb0cf0f9904404da98cfff49a45
   12. 18dcad5ec6bd8a40cf8fc3892c6d401224057c4b

rogyar committed:
Mark test as incomplete because of random Bamboo fails

magento-engcom-team committed:
ENGCOM-3007: Mutations coupons #163

 - Merge Pull Request magento/graphql-ce#163 from magento/graphql-ce:mutations-coupons
 - Merged commits:
   1. ced23016cc71ae62093aa26b912eee378a70549a
   2. 6abc7e56130d24588263ae7e4aa777ba24190c2c
   3. b6eaa1fac65125da8a43cd8fe8a825c55a1f9e4f
   4. 1a077ac7edb2446152c8842cebe45b0782a33676
   5. 9cb459e732d492bb9009e55be42db77dfe520b01
   6. ba6dd12a2d68010045905467bb4a15ec447538a8
   7. 02c2bc038666985e9dc357adba196269d18bbab5
   8. 7f1148fccf62dd7ee1d2adaf4e709a639ae2e135
   9. b6365e197588edc192623ee4e7825fd3e4086548
   10. 3007b3a2fcb2222b5dc405c4273c2a63bf3dd77e
   11. 2a0a450aa937533fc1c9c4da2268515a894f9018
   12. 462601c1c46a731d7549c7d82af70ee8605174fa
   13. 2f4767d72014a359e36f16abc85b68e889e81c1c
   14. 343e4639eca583849cd38107596b5c7a97ff6ef7
   15. 46b40b4b80e6fbffbccd6d95df102fe0dd3f9a6b
   16. fc12cb735355818646aa11cd7340487840cd7abd

Commit:
ENGCOM-3007: Mutations coupons #163

Commit:
ENGCOM-3004: [Recheck] Allow changing logged in customers password with Graphql mutation #187

Commit:
Merge pull request #3188 from magento-engcom/graphql-develop-prs

[EngCom] Public Pull Requests - GraphQL

Commit:
Merge branch 'PR18087' into 2.3-develop

Commit:
2.3-develop fix based on reviews
